### PR TITLE
Add passport support

### PR DESCRIPTION
**Pull request by ianwremmel**
This PR adds support for [passportjs](http://www.passportjs.org/). it got delayed for a while and probably needs to be updated, but should be reasonably easy to adapt to the current code base if so desired.

### CODE REVIEW

**Comment by ianwremmel** [comment]
Note to self: need a way to add the Spark instance to the request

File: packages/passport-ciscospark/test/automation/fixtures/router.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  done(null, `id`);
+});
+
+passport.deserializeUser((id, done) => {
+  done(null, userDB);
+});
+
+passport.use(new CiscoSparkStrategy({
+  // eslint-disable-next-line camelcase
+  redirect_uri: `http://127.0.0.1:${process.env.PORT}`
```
Comment (adamweeks) [inline packages/passport-ciscospark/test/automation/fixtures/router.js]: Do we need to use the redirect_uri from options here?

Comment (ianwremmel) [inline packages/passport-ciscospark/test/automation/fixtures/router.js]: yea, i think this was a remnant from an earlier iteration. I'll fix it.

**Comment by ianwremmel** [comment]
marking as blocked for now; should be revisited after the auth overhaul

**Pull request closed without merging**